### ISSUE TITLE
0.10.x breaks -I input for *.sil classes

### ISSUE THREAD

**Issue by jvgaultney** [sile-typesetter/sile#798]
SILE no longer recognizes *.sil classes specified on the command line using -I. For example the docbook examples given on page 9 of the docs fails with the errors below. This also breaks other packages that use the -I command, such as [ftml-sile](https://github.com/silnrsi/ftml-sile).

```
sile -I docbook examples/article-template.xml
This is SILE v0.10.1
Loading docbook

! Unable to find file docbook
	no file 'classes/docbook'
	no file 'docbook'
	no file 'examples//classes/docbook'
	no file 'examples//docbook' at 	<nowhere>


! Unable to find file examples/article-template.xml
	no file 'examples/article-template.xml'
	no file 'examples//examples/article-template.xml' at 	examples/article-template.xml


! Could not find file at 	examples/article-template.xml
```

**Comment by alerque** [comment]
This is not a bug (in spite of _my_ first reaction too!), it is part of the documented breaking change in the way file name guessing works in v0.10.0 and beyond. The example in the manual should be updated to reflect this, but the change itself is a known breaking changes between versions. There really wasn't a way to gracefully switch between behaviors.

The correct usage is to include the file name extension yourself:

    sile -I docbook.sil examples/article-template.xml

**Comment by alerque** [comment]
Just a heads up @jvgaultney some future release of SILE is going to break the class system entirely. We're in the process of changing out the library used for providing an object oriented object model in Lua from *stdlib* to *Penlight*, and SILE's class system uses object inheritance to surface features from base classes when defining a new class.

I will try to make the migration as smooth as possible, but it will almost certainly require updating any custom classes to match. I've made a note of your ftml project as an example case. Are there other projects that use custom classes I should be aware of to test and/or help migrate when the time comes?

**Comment by simoncozens** [comment]
On the changes to the class system, see https://github.com/sile-typesetter/sile/wiki/STD-to-Penlight-Migration

**Comment by jvgaultney** [comment]
Fontproof and ftml-sile are the only ones we've developed.

**Comment by alerque** [comment]
This is more broken than I realized, the example incantation worked for the example doc but it does not work for anything else when the SILE source tree isn't the CWD.

**Comment by alerque** [comment]
This issue can (finally) be fixed after #645 merges. The addition of `loadkit` gives us a way to robustly handle path searching and loading for file extensions other that `*.lua`.

**Comment by alerque** [comment]
Having reviewed this a bit further, I don't think the `-I` option is even fixable. It tries to be too smart and as a result can't easily be manipulated to do any specific thing.

* Does it load Lua code? or process SILE content?
* Does it run before loading the parser or after?
* Does it search for files relative to the PWD? Or the input document? Or the Lua search path?

Some of these combinations are mutually exclusive and guessing can lead to breakable. In #1373 I propose deprecating it entirely in favor of more specific and predictable alternatives.
